src/pages/wallet-overview/wallet-overview.spec.ts

```diff
@@ -23,17 +38,239 @@ describe("WalletOverviewPage", () => {
       providers: [
-        { provide: WalletService, useClass: MockWalletService },
-        { provide: NavController, useClass: MockNavController },
-        { provide: NavParams, useValue: new MockNavParams() }
+        { provide: WalletService, useValue: walletService },
+        { provide: NavController, useValue: navController },
+        { provide: NavParams, useValue: new MockNavParams() },
+        { provide: ToastController, useValue: toastController },
+        { provide: LoadingController, useValue: loadingController }
       ]
     }).compileComponents();
   }));
 
-  // beforeEach(() => {
-  //   fixture = TestBed.createComponent(WalletOverviewPage);
-  //   comp = fixture.componentInstance;
+  beforeEach(() => {
+    fixture = TestBed.createComponent(WalletOverviewPage);
+    comp = fixture.componentInstance;
+  });
+
+  it("should create component", () => expect(comp).toBeDefined());
+
+  it("should have picked currency as undefined", () => {
+    expect(comp.pickedCurrency).toBeUndefined();
+  })
+
+  it("should have dougnutchart as undefined", () => {
+    expect(comp.doughnutChart).toBeUndefined();
+  })
+
+  it("should have no wallets", () => {
+    expect(comp.wallets.length).toEqual(0);
+  })
+
+  it("should have an empty amounts for the dougnut chart", () => {
+    expect(comp.currenciesForDoughnutCanvas.length).toEqual(0);
+  })
+
+  it("should have empty currencies for the dougnut chart", () => {
+    expect(comp.currenciesForDoughnutCanvasCurrencies.length).toEqual(0);
+  })
+
+  it("should have current wallet as undefined ", () => {
+    expect(comp.currentWallet).toBeUndefined();
+  })
+
+  it("should have current wallet index as 0", () => {
+    expect(comp.currentWalletIndex).toEqual(0);
+  })
+
+  it("should have an empty legend list", () => {
+    expect(comp.legendList.length).toEqual(0);
+  })
+
+  it("should have no available currencies", () => {
+    expect(comp.availableCurrencies.length).toEqual(0);
+  })
+
+  it("should have show fund status to be truthy", () => {
+    expect(comp.showFundsStatus).toBeTruthy();
+  })
+
+  it("should have two factor authentication to be falsy", () => {
+    expect(comp.twoFactorStatus).toBeFalsy();
+  })
+
+  it("should have funds visibility standard to 'shown'", () => {
+    expect(comp.walletFundsVisibility).toEqual("shown");
+  })
+
+  it("should have visibility hidden after clicking the show funds switch", () => {
+    comp.fundsSwitch();
+
+    expect(comp.walletFundsVisibility).toEqual("hidden");
+  })
+
+  it("should have show funds boolean to be falsty after clicking the show funds switch", () => {
+    comp.fundsSwitch();
+
+    expect(comp.showFundsStatus).toBeFalsy();
+  })
+
+  it("should show two factor authentication to be falsy after clicking the two factor switch", () => {
+    comp.twoFactorStatusSwitch();
+
+    expect(comp.twoFactorStatus).toBeTruthy();
+  })
+
+  it("should return null after clicking the delete wallet click without wallets", () => {
```

Comment:
Minor detail: in these unit tests you usually do not refer to UI terms like 'click'. Detecting clicks is more E2E territory. Instead simply say something like "Should return null when trying to delete a wallet when there are no wallets".